spec/addressable/uri_spec.rb

```diff
@@ -5490,6 +5490,31 @@ def to_s
   end
 end
 
+describe Addressable::URI, "when given the tld " do
+  it "'uk' should have a tld of 'uk'" do
+    uri = Addressable::URI.parse("http://example.com")
+    uri.tld = "uk"
+
+    expect(uri.tld).to eq("uk")
+  end
+
+  context "witch " do
```

Comment:
Presumably you meant "which"?

Reply:
Yup. Right, `which`, sorry.

`when given tld, which contains a subdomain`

I can by analogy make the method `url.basename=` and in the new PR correct this word, if you do not mind.

Thank you very much for the merge